keydates-worker/keydates_worker.py

```
@@ -488,6 +498,72 @@ def appget(method, params):
+    """event id -> IANA timezone from EVENTS_URL, fetched once per run. Any
+    failure degrades to {} (every post read in UTC) with a log line, never a
+    held run: a missing zone costs at worst one day of skew on evening posts,
+    which is exactly what the worker did before CON-50."""
+    global _event_tz_cache
+    if _event_tz_cache is not None:
+        return _event_tz_cache
+    req = urllib.request.Request(EVENTS_URL, headers={"User-Agent": APPVIEW_USER_AGENT})
+    out = {}
+    for _ in range(3):
+        try:
+            with urllib.request.urlopen(req, timeout=20) as r:
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)